Cryptogravisor.java

```diff
@@ -1,43 +1,77 @@
     
-    private ArrayList<Encryption> encryptlist;
-    private Contacts;
-    private Cryptogravisor() {
+    private ArrayList<Encryption> encryptlist = new ArrayList<Encryption>();
+    private Contacts contacts = new Contacts();
+    private Communications comm;
+    public Cryptogravisor() {
+	super ("ayylmao");
 	//initialize GUI and stuff
+	comm=new Communications(6000,6001,this);
+	setVisible(true);
     }
     
-    private void actionPerformed() {
+    public void actionPerformed() {
 	//responds to user input
 	//delegates message sends, and user adds
 
     }
     private void addToContacts(String name, String address) {
-	//adds the user entry to the contacts class
+    	// Creates the big integer
+    	BigInteger bigint = Encryption.calcMod();
+    	// adds the user entry to the contacts class
+    	contacts.addContact(name, address);
+    	encryptlist.add(new Encryption(bigint));
+    	// sends the key and mod to given address 
+    	comm.sendKeyAndMod(address, encryptlist.get(encryptlist.indexOf(address)).DiffieHellmanComputeKey(), bigint);
     }
     private void updateContactList() {
 	//updates GUI, to the current contents of Contacts class
     }
-    private void sendMSG() {
+    private void sendMSG(String address, String message) throws InvalidatedEncryptionException {
 	//sends messaged based on the user's selected contact and their message in the textfield
 	//encrypts for recipient, and sends to user with communications class
+    	// sends the encrypted message to the address
+	    comm.send(address, encryptlist.get(encryptlist.indexOf(address)).encryptText(message));
     }
-    public void handleMessage(String message, String address) {
-	
+    public void handleMessage(String message, String address) throws InvalidatedEncryptionException {
 	//called by server, and will be printed to GUI by this method
+	    	// Creates the variables to find the correct encryption address
+	    	int index = contacts.indexOfAddress(address);
+	    	// When message is received it will be encrypted prints out encrypted message for testing purposes
+	    	System.out.print(message);
+	    	// Decrypts the message and prints it
+	    	// Assumes all messages that are received are encrypted
+	    	System.out.print(encryptlist.get(index).decryptText(message));
     }
-    public void handleKey(Object info, String address) {
+    public void handleKey(byte[] info, String address) {
 	//handles encryption info passed for public key crypt
+    	encryptlist.add(new Encryption(Encryption.calcKey(info)));
+    	addToContacts("New Friend", address);
     }
 	
     public void quit() {
 	//ends program, sever connections
     }
+    
+    public void handleKeyAndMod(byte dat[], byte mod[], String address){
+    	// Adds a new friend contact for the user to change the name of
+    	addToContacts("New Friend", address);
+    	// Sends the key
+    	comm.sendKey(address, new Encryption(mod).DiffieHellmanComputeKey());
```

Comment:
save the new Encryption instance to a LOCAL variable, so you can add it to the arraylist. Other it wont be added in the arraylist